sound/soc/sof/core.c

```diff
@@ -324,13 +327,16 @@ static int sof_probe(struct platform_device *pdev)
 		goto comp_err;
 	}
 
+/* bypass DSP if in force legacy hda debug mode */
+#ifndef CONFIG_SND_SOC_SOF_FORCE_LEGACY_HDA
```

Comment:
make this a variable that is enabled if IS_ENABLED(FORCE_LEGACY_HDA) || (DSP not enabled)

Reply:
OK, will change it.